### PR TITLE
Removed web_users_by_domain couch view and a few unused views

### PR DESCRIPTION
**Pull request by orangejenny**
## Summary
Noticed that there are a few unused couch views in `users_extra` and that the `web_users_by_domain` view can be replaced by alternative calls in `users.dbaccessors`.

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

None, I don't think.

### QA Plan

Not requesting QA.

### Safety story
This is primarily removing dead code. I tested the scheduled reports view and the analytics task locally.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/cachehq/cachemodels.py
Comment (millerdev) [inline corehq/apps/cachehq/cachemodels.py:27]: I think removing these (well actually, removing the associated `map.js` and `reduce.js` files) but not removing all other "users" views (i.e., not removing the entire ` corehq/apps/users/_design` directory) means Couch will rebuild all other user views. Do we want that to happen?

This understanding of how Couch works is based on what @dannyroberts told me a few days ago. Maybe I misunderstood?

Comment (orangejenny) [reply corehq/apps/cachehq/cachemodels.py:27]: Yeah, my understanding is that both `users` and `users_extra` will be reindexed, and Danny is the person who best knows how burdensome that is.

Comment (esoergel) [reply corehq/apps/cachehq/cachemodels.py:27]: If we do remove these views, we might consider moving the remaining views each into their own design doc (in `corehq/couchapps`), assuming that's not much more effort.  That way we can subsequently delete say `domain/old_users` without needing to rebuild the rest.

Comment (orangejenny) [reply corehq/apps/cachehq/cachemodels.py:27]: Hi @dannyroberts , any thoughts on this?

Comment (dannyroberts) [reply corehq/apps/cachehq/cachemodels.py:27]: Yeah I think it would be quite burdensome if it were in the main db, but I'm guessing this is only in the commcarehq__users db, right? That has about 160k documents, which not nothing but also not out of the question. Enough that we might want to do it on a weekend unless we had a better way to estimate it.

I'd also ask that we hold off on merging this until a week or so after we've upgraded to couch 3 just to space out any small amount of risk there might be.

Comment (orangejenny) [reply corehq/apps/cachehq/cachemodels.py:27]: Yeah, this should just be commcarehq__users. Happy to hold off until couch 3 has been live for a bit.

Comment (orangejenny) [reply corehq/apps/cachehq/cachemodels.py:27]: I believe Couch 3 has been live for a while now, any objections to merging it this Friday?

@dannyroberts @skodde @calellowitz

Comment (dannyroberts) [reply corehq/apps/cachehq/cachemodels.py:27]: No objections here!

Comment (snopoke) [reply corehq/apps/cachehq/cachemodels.py:27]: @orangejenny @dannyroberts should we hold off a little bit longer until I remove the `roles_by_domain` view (hopefully in the next 2 weeks)?

Comment (orangejenny) [reply corehq/apps/cachehq/cachemodels.py:27]: Sure, works for me.

Comment (snopoke) [reply corehq/apps/cachehq/cachemodels.py:27]: https://github.com/dimagi/commcare-hq/pull/30179

Comment (orangejenny) [reply corehq/apps/cachehq/cachemodels.py:27]: @snopoke Is https://github.com/dimagi/commcare-hq/pull/30179 (and https://github.com/dimagi/commcare-hq/pull/30178) ready to merge today?

Comment (snopoke) [reply corehq/apps/cachehq/cachemodels.py:27]: #30178 can be merged but I was going to wait before removing the couch view otherwise rollback required rebuilding the view

Comment (orangejenny) [reply corehq/apps/cachehq/cachemodels.py:27]: okay, so should I hold off on merging this PR and preindexing until next week? (there's no hurry)

Comment (snopoke) [reply corehq/apps/cachehq/cachemodels.py:27]: yes, thanks. I've merged the other PR so that will go out in the next deploy. Once that has been live for a week we can merge these.

Comment (orangejenny) [reply corehq/apps/cachehq/cachemodels.py:27]: cool, sounds good

Comment (orangejenny) [reply corehq/apps/cachehq/cachemodels.py:27]: postponing a week because the Thursday morning deploy didn't go through, so https://github.com/dimagi/commcare-hq/pull/30178 isn't live yet

Comment (orangejenny) [reply corehq/apps/cachehq/cachemodels.py:27]: actually postponing a few weeks because I'm off the next 2 Fridays. @snopoke if you want to merge this and preindex next Friday go for it, and if you don't get to it, I will eventually.

**Review by snopoke — APPROVED** [review]

**Pull request merged**
